app/models/notification.rb

```diff
@@ -0,0 +1,31 @@
+class Notification < ActiveRecord::Base
+  acts_as_readable on: :updated_at
+  belongs_to :user
+  belongs_to :course
+
+  scope :center_popup, -> { where(type: 'Notification::CenterPopup') }
+  scope :right_side_popup, -> { where(type: 'Notification::RightSidePopup') }
+  scope :email, -> { where(type: 'Notification::EmailNotification') }
+
+  def self.notify(user, course, options = {})
+    case options[:type]
```

Comment:
I think this should accept an array of methods to notify by. I might want to push to the activity feed and as a popup at the same time. Also, maybe a Facebook post.